draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1581,75 +1581,121 @@ Steps:
 26. return (x, y)
 ~~~
 
-## Mappings for twisted Edwards curves
+## Mappings for Edwards curves
 
-### Elligator 2 Method {#ell2edwards}
+Edwards curves are closely related to Montgomery
+curves ({{montgomery}}): every Edwards curve is birationally equivalent
+to a Montgomery curve ({{BBJLP08}}, Theorem 3.2).
+This equivalence yields an efficient way of hashing to an Edwards curve:
+first, hash to the equivalent Montgomery curve, then transform the
+result into a point on the Edwards curve by evaluating a pair of rational
+functions called a birational map.
+Thus, hashing to an Edwards curve requires first identifying a
+corresponding Montgomery curve and birational map.
+
+### Birational maps from Montgomery to Edwards curves {#birational-map}
+
+There are two ways to identify the Montgomery curve and birational map
+for use with a given Edwards curve.
 
-Twisted Edwards curves and Montgomery curves are closely related: every
-twisted Edwards curve is birationally equivalent to a Montgomery curve
-({{BBJLP08}}, Theorem 3.2). To hash to a twisted Edwards curve, hash
-to the equivalent Montgomery curve and evaluate the rational map to
-obtain a point on the twisted Edwards curve.
+When hashing to a standard Edwards curves for which a corresponding
+Montgomery form and birational map are also standardized, the standard
+Montgomery form and birational map MUST be used to ensure compatibility
+with existing software.
+Two examples of this are the edwards25519 and edwards448 curves,
+which correspond to the Montgomery curves curve25519 and curve448, respectively.
+For both of these curves, {{RFC7748}} lists both the Montgomery and Edwards
+forms and gives the corresponding rational maps.
 
-For a twisted Edwards curve given by a * x^2 + y^2 = 1 + d * x^2 * y^2,
-first compute A and B, the parameters of the equivalent Montgomery curve,
-as follows:
+Some Edwards curves do not have a standardized Montgomery form.
+In this case, the following procedure MUST be used to derive a birational map.
```

Comment:
nit: this only derives one of the maps, right? (it seems left as an exercise for the reader to describe the reverse mapping)

Reply:
Since we don't need the inverse map for this hash, I wasn't going to bother with it.

But it's pretty short to specify the inverse, and could make life easier in the future. I'll add it.

Reply:
I'm OK with either, so this works for me.